srml/support/procedural/src/storage/impls.rs

```diff
@@ -481,4 +481,74 @@ impl<'a, I: Iterator<Item=syn::Meta>> Impls<'a, I> {
 			}
 		}
 	}
+
+	pub fn double_map(self, k1ty: &syn::Type, k2ty: &syn::Type, k2_hasher: TokenStream2) -> TokenStream2 {
+		let Self {
+			scrate,
+			visibility,
+			traitinstance,
+			traittype,
+			type_infos,
+			fielddefault,
+			prefix,
+			name,
+			attrs,
+		} = self;
+		let DeclStorageTypeInfos { typ, value_type, is_option, .. } = type_infos;
+		let option_simple_1 = option_unwrap(is_option);
+
+		let as_double_map = quote!{ <Self as #scrate::storage::unhashed::generator::StorageDoubleMap<#k1ty, #k2ty, #typ>> };
+
+		let mutate_impl = if !is_option {
+			quote!{
+				#as_double_map::insert(key1, key2, &val, storage)
+			}
+		} else {
+			quote!{
+				match val {
+					Some(ref val) => #as_double_map::insert(key1, key2, &val, storage),
+					None => #as_double_map::remove(key1, key2, storage),
+				}
+			}
+		};
+		// generator for double map
+		quote!{
+			#( #[ #attrs ] )*
+			#visibility struct #name<#traitinstance: #traittype>(#scrate::storage::generator::PhantomData<#traitinstance>);
+
+			impl<#traitinstance: #traittype> #scrate::storage::unhashed::generator::StorageDoubleMap<#k1ty, #k2ty, #typ> for #name<#traitinstance> {
+				type Query = #value_type;
+
+				fn prefix() -> &'static [u8] {
+					#prefix.as_bytes()
+				}
+
+				fn key_for(k1: &#k1ty, k2: &#k2ty) -> Vec<u8> {
+					let mut key = #as_double_map::prefix_for(k1);
+					key.extend(&#scrate::Hashable::#k2_hasher(k2));
```

Comment:
Here we may want to change the scheme : just concatenating both key will result in db conflict see #2002 comment, so full key should be something like twox(key1) + (keccak(key2) xor twox(key1)) . 
If we consider twox to be colision resistant there is probably no need for xor on the remaining bit of key1.
Not sure it should be done in this PR or in an ulterior one (just a good thing to keep it in mind).

Reply:
It is not a good idea anymore to do it (a pending trie pr will solve this issue by prepending trie path in key). It may not have been a good idea to do it in this pr anyway.

Reply:
ok thanks for looking for this kind of stuff, because I'm not really aware of that for now